filters/include/pcl/filters/farthest_point_sampling.h

```diff
@@ -0,0 +1,127 @@
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#pragma once
+
+#include <pcl/filters/filter_indices.h>
+#include <climits>
+#include <random>
+namespace pcl
+ {
+    /** \brief @b FarthestPointSampling applies farthest point sampling using euclidean 
+      * distance, starting with a random point, utilizing a naive method.
+      * \author Haritha Jayasinghe
+      * \ingroup filters
+      * @TODO: add support to export/import distance metric
+      */
+    template<typename PointT>
+    class FarthestPointSampling : public FilterIndices<PointT>
+    {
+      using PCLBase<PointT>::input_;
+      using PCLBase<PointT>::indices_;
+      using Filter<PointT>::filter_name_;
+      using FilterIndices<PointT>::keep_organized_;
+      using FilterIndices<PointT>::user_filter_value_;
+      using FilterIndices<PointT>::extract_removed_indices_;
+      using FilterIndices<PointT>::removed_indices_;
+    
+      using typename FilterIndices<PointT>::PointCloud;
+
+      public:
+        /** \brief Empty constructor. */
+        FarthestPointSampling (bool extract_removed_indices = false) : 
+          FilterIndices<PointT> (extract_removed_indices),
+          sample_ (std::numeric_limits<int>::max ()), 
+          seed_ (static_cast<unsigned int> (std::random_device()()))
+        {
+          filter_name_ = "FarthestPointSamping";
+        }
+
+        /** \brief Set number of points to be sampled.
+          * \param sample
+          */
+        inline void
+        setSample (std::size_t sample)
+        {
+          sample_ = sample;
+        }
+
+        /** \brief Get the value of the internal \a sample_ parameter.
+          */
+        inline std::size_t
+        getSample ()
+        {
+          return (sample_);
+        }
+
+        /** \brief Set seed of random function.
+          * \param seed
+          */
+        inline void
+        setSeed (unsigned int  seed)
+        {
+          seed_ = seed;
+        }
+
+        /** \brief Get the value of the internal \a seed_ parameter.
+          */
+        inline unsigned int 
+        getSeed ()
+        {
+          return (seed_);
+        }
+
+      protected:
+
+        /** \brief Number of points that will be returned. */
+        std::size_t sample_;
```

Comment:
Rename it to `sample_size_` ?

Reply:
Yeah that makes more sense, but I recommend chaging it across all the other filters as well since other algorithms that select a specified number of points (such as random sampling, normal space) use the same `sample_` naming.